app/lib/data/common_services/models/network/api_response.dart

```diff
@@ -0,0 +1,23 @@
+abstract class ApiResponse<T> {}
+
+class Success<T> extends ApiResponse<T> {
+  Success({required this.data});
+
+  T? data;
+}
+
+class Failure<T> extends ApiResponse<T> {
+  Failure({required this.failureType, this.error});
+  FailureType failureType;
+
+  /// if status code is none of 400, 401, 403, 500
+  /// we return status code only as error
+  /// otherwise, leave it null
+  String? error;
+}
+
+enum FailureType {
+  unknown,
+  badRequest,
+  noAuthorization,
```

Comment:
Cann you please add documentation here, like:

```suggestion
   /// Represents http errors 401, 403.
  noAuthorization,
  /// Unknown error.
  unknown, // <-- usually unknown is at the end of such an enum.
```